### ISSUE TITLE
XdocsExamplesAstConsistencyTest does not detect difference in method parameters

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#19458]
Detected at https://github.com/checkstyle/checkstyle/pull/19453/changes#diff-b004bea1a8e7b4330157ecb1e2b23173833d444dd6eb0068800ed72e26ccb846R16

src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/parametername/Example1.java

Has `public boolean equals(Object v3) {` but other has `public boolean equals(Object V3)`

It should be violation.

Any difference except for trailing comments should be violation.

**Comment by romani** [comment]
@surajgojanur, can you help us to fix this gap ?

**Comment by surajgojanur** [comment]
> [@surajgojanur](https://github.com/surajgojanur), can you help us to fix this gap ?

yes i'm on it

**Comment by smita1078** [comment]
This issue has now been resolved as part of the work in **#20284**.

The AST consistency check was extended to detect the previously missed inconsistencies by:

* including `IDENT` tokens in the structural comparison (while continuing to ignore class and constructor names via the existing exception), and
* comparing section-relative line numbers to ensure structural equivalence between examples.

With these changes, differences such as:

```java
public boolean equals(Object v3)
```

vs.

```java
public boolean equals(Object V3)
```

are now correctly reported as violations. As intended, any difference other than trailing comments is treated as an inconsistency.
